lib/LedgerNodePeers.js

```diff
@@ -0,0 +1,145 @@
+/*!
+ * Ledger node operations management class.
+ *
+ * Copyright (c) 2017-2020 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const bedrock = require('bedrock');
+const database = require('bedrock-mongodb');
+const {util: {BedrockError}} = bedrock;
+const {COLLLECTION_NAME} = require('./ledgerNodePeerCollection');
+
+/**
+ * The LedgerNodePeers class exposes peer information.
+ */
+module.exports = class LedgerNodePeers {
+  constructor({ledgerNode}) {
+    this.ledgerNode = ledgerNode;
+    this.ledgerNodeId = ledgerNode.id;
+  }
+
+  async add({peer}) {
+    const collection = database.collections[COLLLECTION_NAME];
+    const now = Date.now();
+    const peerRecord = {
+      meta: {
+        created: now,
+        ledgerNodeId: this.ledgerNodeId,
+        updated: now
+      },
+      peer,
+    };
+    let result;
+    try {
+      result = await collection.insertOne(peerRecord);
+    } catch(e) {
+      if(database.isDuplicateError(e)) {
+        throw new BedrockError(
+          'Duplicate peer.',
+          'DuplicateError', {
+            httpStatusCode: 409,
+            public: true,
+            ledgerNodeId: this.ledgerNodeId,
+            peerId: peer.id,
+          }, e);
+      }
+    }
+
+    return result.ops[0];
+  }
+
+  async get({id}) {
+    const collection = database.collections[COLLLECTION_NAME];
+    const query = {
+      'meta.ledgerNodeId': this.ledgerNodeId,
+      'peer.id': id,
+    };
+    const result = await collection.findOne(query, {_id: 0, peer: 1});
+
+    if(!result) {
+      throw new BedrockError('Peer not found.', 'NotFoundError', {
+        httpStatusCode: 404,
+        ledgerNodeId: this.ledgerNodeId,
+        peerId: id,
+        public: true,
+      });
+    }
+
+    return result.peer;
+  }
+
+  async getAll() {
+    const collection = database.collections[COLLLECTION_NAME];
+    const query = {
+      'meta.ledgerNodeId': this.ledgerNodeId,
+    };
+    const result = await collection.find(query, {
+      projection: {_id: 0, peer: 1}
+    }).toArray();
+
+    const peers = result.map(r => r.peer);
+
+    return peers;
+  }
+
+  async getRecommended() {
```

Comment:
this is basically a duplicate method for getAll in addition to the fact that we should probably allow getAll to take selective queries.

Reply:
Your observation is correct, however, a convenience API for this hot path is desired here.  Also, we currently have no use case for custom queries for peers, so, in this case, that will not be a 1.0 feature, but certainly may be added later.